src/main/java/com/thebois/models/beings/Pawn.java

```diff
@@ -0,0 +1,62 @@
+package com.thebois.models.beings;
+
+import com.thebois.models.Position;
+
+/**
+ * Pawn.
+ */
+public class Pawn implements IBeing {
+
+    private final Position currentPosition;
+    private Position destination;
+
+    /**
+     * Creates a Pawn.
+     *
+     * @param currentPosition the initial position of the pawn.
+     */
+    public Pawn(Position currentPosition) {
+        this.currentPosition = currentPosition;
+    }
+
+    // proof of concept, preliminary function
+    private void move() {
+        final float maxWalkingDistance = 1;
+
+        // Calculate delta of distance between current position and the destination
+        final float deltaX = this.destination.getPosX() - this.currentPosition.getPosX();
+        final float deltaY = this.destination.getPosY() - this.currentPosition.getPosY();
+
+        // Decide whether to walk the maximum distance or to walk a shorter distance
+        final float walkDistanceY = Math.min(Math.abs(maxWalkingDistance), Math.abs(deltaY));
+        final float walkDistanceX = Math.min(Math.abs(maxWalkingDistance), Math.abs(deltaX));
+
+        // Calculate direction of the distance to walk
+        final float distanceX = Math.signum(deltaX) * walkDistanceX;
+        final float distanceY = Math.signum(deltaY) * walkDistanceY;
+
+        // Calculate new position
+        final float newPosX = this.currentPosition.getPosX() + distanceX;
+        final float newPosY = this.currentPosition.getPosY() + distanceY;
```

Comment:
Maybe this is just me but this would be a lot simpler using vector math.

Reply:
You should also factor in `deltaTime` when updating the position. Right now the movement speed is dependent on framerate. Using `deltaTime` (the amount of time between the previous frame and the current) movement speed will be independent of framerate.

Reply:
No need to refactor any temporary stuff to use vectors though, those are fine since they're temporary.

Reply:
I will leave this for now. We can improve it when we have a better idea of how it could be implemented, like you say.